src/libraries/System.Diagnostics.Process/tests/ProcessTestBase.cs

```diff
@@ -13,7 +13,7 @@ namespace System.Diagnostics.Tests
 {
     public partial class ProcessTestBase : FileCleanupTestBase
     {
-        protected const int WaitInMS = 30 * 1000;
+        protected readonly int WaitInMS = 30 * 1000 * (PlatformDetection.IsReleaseRuntime ? 1 : 3);
```

Comment:
Is this needed both Checked and Debug, or just for Debug runtime?

Reply:
> Is this needed both Checked and Debug, or just for Debug runtime?

Good point -- I would have to try.

> The debug runtime is super slow. We never run tests on it.
> 
> If we want to be hardening the tests for super slow modes of execution (GCStress is another example of that), we should introduce a concept timeout multiplier and apply it to all hardcoded timeouts.

In this case, I see the change above fixed one of the failing tests, but not the other (ie.,  even 90 sec was not even enough).

I had figured as I have a spare machine, it might be interesting to occasionally run the tests on a debug runtime. However I don't want to go through all hardcoded timeouts in the tree, since they're in many places and as you suggest there's better uses of time. What about if I simply add something like `public static int TimeoutMultiplier = (PlatformDetection.IsDebugRuntime ? 1 : 5)` into PlatformDetection, and apply it opportunistically as I see failures in my local run?

Reply:
Then, if and when I play with other slow modes like GCStress, I can extend TimeoutMultiplier to incorporate those. That right now is only understood in [SkipOnCoreClrDiscoverer](https://github.com/danmoseley/arcade/blob/main/src/Microsoft.DotNet.XUnitExtensions/src/Discoverers/SkipOnCoreClrDiscoverer.cs#L16).

If Checked is slower than Release but faster than Debug, TimeoutMultiplier could also potentially handle that.

Reply:
> What about if I simply add something like public static int TimeoutMultiplier = (PlatformDetection.IsDebugRuntime ? 1 : 5) into PlatformDetection

Yes, that sounds good.